common/src/main/java/org/vivecraft/client_vr/VRData.java

```diff
@@ -96,35 +100,33 @@ public VRData(Vec3 origin, float walkMul, float worldScale, float rotation) {
             Matrix4f scopeOff = this.getSmoothedRotation(1, 0.2F);
             this.t0 = new VRDevicePose(this,
                 scopeMain,
-                mcVR.getAimSource(0).add(scaleOffset),
-                scopeMain.transform(Vector3.forward()).toVector3d());
+                mainAimSource,
+                scopeMain.transformDirection(MathUtils.BACK, new Vector3f()));
             this.t1 = new VRDevicePose(this,
                 scopeOff,
-                mcVR.getAimSource(1).add(scaleOffset),
-                scopeOff.transform(Vector3.forward()).toVector3d());
+                offAimSource,
+                scopeOff.transformDirection(MathUtils.BACK, new Vector3f()));
         }
 
         // screenshot camera
-        Matrix4f camRot = Matrix4f.multiply(Matrix4f.rotationY(-rotation),
-            new Matrix4f(dataHolder.cameraTracker.getRotation()).transposed());
-        float inverseWorldScale = 1.0F / worldScale;
+        Matrix4f camRot = new Matrix4f().rotationY(-rotation).mul(dataHolder.cameraTracker.getRotation().get(new Matrix4f()));
         this.cam = new VRData.VRDevicePose(this,
             camRot,
-            dataHolder.cameraTracker.getPosition().subtract(origin).yRot(-rotation).scale(inverseWorldScale)
-                .add(scaleOffset),
-            camRot.transform(Vector3.forward()).toVector3d());
+            dataHolder.cameraTracker.getRoomPosition(origin)
+                .rotateY(-rotation).div(worldScale).add(scaleOffset),
+            camRot.transformDirection(MathUtils.BACK, new Vector3f()));
 
         // third person camera
         if (mcVR.mrMovingCamActive) {
             this.c2 = new VRDevicePose(this,
                 mcVR.getAimRotation(2),
-                mcVR.getAimSource(2).add(scaleOffset),
+                mcVR.getAimSource(2).add(scaleOffset, new Vector3f()),
```

Comment:
Put it next to `mainAimSource ` and `offAimSource`, as a third "controller".

Reply:
after the hand data, or between c and h?

Reply:
I was thinking line 75, when c0 and c1 are done. Might also put c2 their to get all controllers next to each other? Though it doesn't really matter, was just wondering if it would make it more clear.

Reply:
well c2 isnt a fixed thing, there is also the option to have the camera at a fixed location, so I would need to move that whole if/else case there

Reply:
Let's just leave it like this then, it's fine.